### ISSUE TITLE
Testing on Unseen data and Validation Issues

### ISSUE THREAD

**Issue by ads9294** [HuguesTHOMAS/KPConv-PyTorch#164]
Hi,
Thanks for the clean code and your work on point cloud segmentation. Currently, I completed training model (**train_S3DIS.py)** on my custom dataset (Large scale point cloud) and got **mean IoU around 0.4 or 40%**. I have below issues regarding testing and validation, please find the details below:
### 1. **Validation**
When I try to validate the model by running 'test_models.py' file with 1 validation file, it is taking too long to predict the labels. Testing is not getting completed even after 20 epochs. I changed  following lines in test_modes.py by seeing your comment in one of the Large scale point cloud testing threads.
![image](https://user-images.githubusercontent.com/36118503/166066061-48de9cc6-26d5-4c87-a79b-db136679457e.png)

![image](https://user-images.githubusercontent.com/36118503/166066177-5c213bc4-4bcc-4311-97d5-c449a78f4bc9.png)

Can you please tell me if I need to change any other values to reduce the validation time?

2. Testing on new data or data with no labels:
I need to test the model on new data which the model has not seen and with no labels file. Can you please tell me what lines to change in test_models.py file besides the below line. 
![image](https://user-images.githubusercontent.com/36118503/166066988-2e744ad5-e970-46c3-8836-6bb3bc02dc3d.png)

Thanks again!!

**Comment by HuguesTHOMAS** [comment]
> Can you please tell me if I need to change any other values to reduce the validation time?

No, not really, it is a good way to make the test faster. You could define a larger `validation_size` here:
But it should not make things much faster, only each test epoch will last more time.

Then if you want faster processing you can always use a larger input radius and first_subsampling_dl, but that would mean restarting the training and maybe having the performance drop a little bit. Optimal input radius and subsampling are very sensible parameters that affect the performances. With a new dataset, you should probably evaluate the performances with different values to find the best one.

> I need to test the model on new data which the model has not seen and with no labels file.

A new dataset file was recently added especially for this purpose: 
https://github.com/HuguesTHOMAS/KPConv-PyTorch/blob/master/datasets/NPM3D.py

You can use this one as an example for the loading of the test data. You will see that multiple files are loaded for the **'test'** set. 

And then yes you need to tell the _test_models.py_ file to use the **'test'** set
